### ISSUE TITLE
Trashed pages with unpublished status and an entry in the page_path_history table don't 404

### ISSUE THREAD

**Issue by adrianbj** [processwire/processwire-issues#1343]
### Short description of the issue

If you have a page in the trash that has unpublished status and it's also in the page_path_history table because it was renamed at some point before it was trashed, it will load on the frontend for logged in users with the trashed path in the URL, eg: http://pwtest.test/trash/1042.1.1_new-renamed/

### Expected behavior

A 404 should be thrown

### Steps to reproduce the issue

1. Create a page
2. Rename the page so that it gets added to the page_path_history table
3. Trash the page
4. Load the URL for the original name of the page

Tested on PW 3.0.169 and 3.0.172

**Comment by ryancramerdesign** [comment]
This should only be the case for superuser or user with access to trash. Are you saying it does this for all logged in users?

**Comment by adrianbj** [comment]
It's not all users, but I still think it's strange behavior because if you try to visit the last URL for the page (before it was trashed), it will properly 404. I don't think a page path history entry should break this, do you?

**Comment by matjazpotocnik** [comment]
@adrianbj is this still an issue?

**Comment by adrianbj** [comment]
Yep, just confirmed.

**Comment by adrianbj** [comment]
I dug into this and found the root cause, and have a tested fix.

### Root cause

When a page is trashed, its `page_path_history` rows are intentionally kept (they're only removed on hard delete via `hookPageDeleted()`), so redirects can resume if the page is restored. The problem is that neither of the two history-lookup paths ever checks whether the matched page is in the trash:

1. **`PagesPathFinder::getPathHistory()`** (the primary request flow since 3.0.186) takes the history match and unconditionally sets a 301 redirect to the page's *current* path — which for a trashed page is `/trash/1234.1.1_page-name/`. Superusers (and users with trash view access) then pass the access check at that URL and the page renders.
2. **`PagePathHistory::hookPageNotFound()`** (the legacy flow, and the one active on the 3.0.169/172 versions I originally reported this on) only checks `$page->viewable()`, which also passes for those users.

The *last* URL before trashing 404s correctly only because `hookPageMoved()` declines to save a history entry when a page moves to/from the trash — hence the inconsistency between the two URLs. The module already expresses this "no trash paths" intent elsewhere (`getVirtualHistoryParent()` explicitly excludes `/trash/`-style paths), so excluding trashed targets at lookup time is consistent with the existing design.

### Suggested fix

**`wire/core/Pages/PagesPathFinder.php`** — in `getPathHistory()`, right after the `if(!$info['id'])` check (~line 1292). `getPathInfo()` already returns the joined `pages.status`, and trashing applies `Page::statusTrash` recursively to descendants, so this check is free:

```php
// if found page is in the trash do not redirect to it (behave as if no match)
if($info['status'] & Page::statusTrash) {
	$this->addMethod('pathHistory', false, 'Found row but page is in trash');
	return false;
}
```

**`wire/modules/Page/PagePathHistory/PagePathHistory.module`** — in `hookPageNotFound()` (~line 1006):

```php
if($page->id && !$page->isTrash() && $page->viewable()) {
```

### Verified on 3.0.271 against a real install

- **Before:** superuser requesting the pre-rename URL of a trashed, unpublished page got `301 → /trash/47151.1.27_page-name/ → HTTP 200` (page rendered), while the pre-trash URL 404'd — exact repro of this issue.
- **After:** the pre-rename URL returns **404** for superuser and guest alike.
- **No regressions:** a renamed (non-trashed) page's old URL still 301s to its new URL, and after restoring the page from the trash its old URL 301-redirects again — so fixing this at lookup time (rather than deleting history rows at trash time) preserves the restore behavior.

One side observation from testing, possibly worth a separate issue: the multi-language branch of `hookPageMoved()` lacks the trash guards that the default-language branch has, so trashing a page on a multi-language site saves a junk history row for the trash-format path (e.g. `/47151.1.27_page-name`) for the non-default language.

**Comment by ryancramerdesign** [comment]
Confirmed and fixed as suggested, reproduced end-to-end before applying: created a page, renamed it (creating a history entry), then trashed it — before the fix, the old URL 301-redirected into `/trash/...` and rendered for a superuser; the pre-trash URL correctly 404'd. Applied both changes and re-verified:

| Scenario | Before | After |
|---|---|---|
| Old URL of a trashed, renamed page | 301 → `/trash/...` (200) | **404** |
| Restoring the page | — | 301 redirect resumes, unaffected |

Fixed in ffbfaedb.

-Claude
